lib/gateway/get-query-result.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+const {
+  parse
+} = require('graphql')
+const { preGatewayExecutionHandler } = require('../handlers')
+
+const mergeQueries = (queries) => {
+  const q = [...new Set(queries.map(q => q.query))]
+  const resultIndexes = []
+  const mergedQueries = queries.reduce((acc, curr, queryIndex) => {
+    if (!acc[curr.query]) {
+      acc[curr.query] = curr.variables
+      resultIndexes[q.indexOf(curr.query)] = []
+    } else {
+      acc[curr.query].representations = [
+        ...acc[curr.query].representations,
+        ...curr.variables.representations
+      ]
+    }
+
+    for (let i = 0; i < curr.variables.representations.length; i++) {
+      resultIndexes[q.indexOf(curr.query)].push(queryIndex)
+    }
+
+    return acc
+  }, {})
+
+  return { mergedQueries, resultIndexes }
+}
+
+const getBactchedResult = async ({ mergeQueriesResult, queries, serviceDefinition, service }) => {
+  const { mergedQueries, resultIndexes } = mergeQueriesResult
+  const context = queries[0].context
+  const originalRequestHeaders = queries[0].originalRequestHeaders
+  const batchedQueries = []
+
+  for (const [query, variables] of Object.entries(mergedQueries)) {
+    const document = parse(query)
```

Comment:
I'm pretty sure we have this parsed already., could we avoid it? I see we have this done in the previous code as well... so if it's too heard to fix here let's open a separate issue to remove it.

Reply:
Found! :)